### DISCUSSION TITLE
How to properly define a design support using C#?"

### DISCUSSION THREAD

**Discussion by lucas-ita** [Dlubal-Software/Dlubal_CSharp_Client#58] [Q&A] [answered]
Hi,

When I manually configure the design support in RFEM and then use SoapUI to get the method's response, I receive the following:

![SoapUI](https://github.com/Dlubal-Software/Dlubal_CSharp_Client/assets/123772761/8a6fc099-a160-436c-87b5-53fd6f1eea4e)

Now, when I try to do the same thing but in an automated way, problems arise. The code below sets design support on members based on comments to determine if they are start or end, then separates each one and sends them to RFEM.

```
 // creates new model
                string modelUrl = application.get_active_model();


                #region new model
                // connects to RFEM6/RSTAB9 model
                ModelClient model = new ModelClient(Binding, new EndpointAddress(modelUrl));

                #endregion

                object_location[] sel_objs = model.get_all_selected_objects();

                List<int> design_supports_start = new List<int>();
                List<int> design_supports_end = new List<int>();
                List<int> design_supports_nodes_start = new List<int>();
                List<int> design_supports_nodes_end = new List<int>();

                foreach (object_location obj in sel_objs)
                {
                    if (obj.type == object_types.E_OBJECT_TYPE_MEMBER)
                    {
                        member memberRFEM = model.get_member(obj.no);
                        

                        if (memberRFEM.comment == "V_START")
                        {
                            design_supports_start.Add(obj.no);
                            design_supports_nodes_start.Add(memberRFEM.node_start);
                        }
                        else if (memberRFEM.comment == "V_END")
                        {
                            design_supports_end.Add(obj.no);
                            design_supports_nodes_end.Add(memberRFEM.node_end);
                        }
                    }
                }

                int[] assigned_to_members = design_supports_start.Concat(design_supports_end).ToArray();
                int[] assigned_to_nodes = design_supports_nodes_start.Concat(design_supports_nodes_end).Distinct().ToArray();
             

                design_support designSupport_GH = new design_support
                {
                    no = 1,
                    assigned_to_members = assigned_to_members,
                    assigned_to_nodes = assigned_to_nodes

                };
              
                model.set_design_support(designSupport_GH);

                model.finish_modification();

```
Here's the outcome of this code in my RFEM model:
![image](https://github.com/Dlubal-Software/Dlubal_CSharp_Client/assets/123772761/b825e7f6-d234-484d-b208-72e7ae909846)

Using SoapUI again to obtain the response from the design support method, but this time from the model where the design support configuration was done automatically, I get the following as a response:

![image](https://github.com/Dlubal-Software/Dlubal_CSharp_Client/assets/123772761/43cf7013-9799-4ea7-a5ff-470f96bbfc86)

Note that the response is the same as when design support was done manually, which is fine so far. However, upon clicking on the member that previously had only the support configuration at the end, it now has configurations at both the beginning and end. Could this be considered a bug?

![image](https://github.com/Dlubal-Software/Dlubal_CSharp_Client/assets/123772761/d2a7259f-822b-4476-aada-485931dd4660)

**Comment by Rebecca-Coding21** [comment, marked as answer]
Hi @lucasngbr-eng,
in order to analyse your problem more precisely it would be necessary to take a look at your model and code. Could you please send your code and model to rebecca.goebel@dlubal.com? I will then analyse you issue and reply to you as soon as possible.
